### PR TITLE
Deploy image using docker

### PR DESCRIPTION
**Pull request by mckennagalvin**
Built the webserver using docker and shrunk it to compile a mostly static binary.
I deployed it on AWS under the public IP 54.190.63.110, but I currently can only curl into the server if I ssh into the machine and send a request to localhost -- I think some kind of security group setting is limiting public access for the server, so I'll look into that!

### CODE REVIEW

**Comment by mckennagalvin** [comment]
Update: got the public IP address to work! I updated the README with some links to test out the server and deployment instructions. I also added a script to automate AWS deployment.

File: Makefile [hunk 1]
```diff
@@ -31,6 +31,18 @@ all: webserver
...
 	g++ $(CXXFLAGS) -I. $(SRC_FILES) -o webserver $(BOOST_FLAGS)
 
+docker:
+	rm -f deploy/webserver
+	docker build -t httpserver.build .
+	docker run httpserver.build > deploy/binary.tar
+	cd deploy && tar -xvf binary.tar
+	rm -f deploy/binary.tar
+	docker build -t httpserver deploy
+	docker run --rm -t -p 8080:8080 httpserver
```
Comment (mattdrag) [inline Makefile]: Maybe we should have different targets for building the docker and deploying it?

**Comment by mattdrag** [comment]
Nice! I checked it out and everything seems to work great!

**Comment by andykuang1** [comment]
Tested it as well with both echo and static files. All ok :ok_hand:

**Review by andykuang1 — APPROVED** [review]

File: Makefile [hunk 2]
```diff
@@ -31,6 +31,18 @@ all: webserver
...
+
+aws:
+	./aws_deploy.sh
+
```
Comment (mattdrag) [inline Makefile:45]: This is great! It's nice to have Makefile targets for building and deploying the server

File: README.md [hunk 3]
```diff
@@ -19,18 +41,23 @@ To test the static file handler, you can try requesting some of the following re
...
 * `make test` builds the test binaries for the individual files. You can then run each test separately, if you choose (for example, `./request_handler_echo_test`)
 * `make all-tests` builds and runs all unit and integration tests
 
+---
 
-###Dealing with bound ports
+### Dealing with bound ports
 If when running the server you get an error that the port is already bound,
 execute `sudo lsof -i:8080` (change 8080 to whatever port number you're using).
 Then find the PID of the process associated with the server and kill it using `sudo kill -9 <pid>`
```
Comment (mattdrag) [inline README.md:56]: I'm pretty sure that CTRL-Z'ing while running the server was causing this, CTRL-C would always kill the process. In any case, this wont be a problem now that the server is deployed

**Pull request merged**